### PR TITLE
One method of simulating Z80 open bus

### LINKED ISSUE #143: Sonic Delta 40Mb sound problem (Z80 interrupt mode 0)
The ROM hack Sonic Delta 40Mb has two different sound drivers. One of the drivers, which seems to handle music from Sonic 3 and Sonic & Knuckles, operates without problems. However, the other driver gets stuck in a loop waiting for an interrupt to occur while interrupts are disabled, causing sound to be distorted or completely absent.

The problem can be observed simply by loading the game, since the sound driver loaded for the Sega logo and title screen is the one with the problem. One way to manually switch between the drivers is to use the Sound Test (enter Up, Up, Down, Down, Up, Up, Up, Up, Up at the title screen menu) and then play sound 33 or lower to load the working driver, or sound 34 or higher to load the non-working driver.

I believe this is because of a difference in behavior with Z80 interrupt mode 0. According to the Z80 manual, a reset changes the interrupt mode to 0, and the driver does not seem to ever change the interrupt mode. In interrupt mode 0, the byte on the data bus during the interrupt cycle is interpreted as an instruction to execute (encoded as an I/O cycle that is neither a read nor a write). According to Neto, the author of the ROM hack, both drivers work normally on a wide range of Sega and Sega-licensed hardware. I think this works on original consoles because the Z80 data bus is pulled up (by what looks like a SIP resistor array in the first schematic I found), and is not being driven because /RD is not asserted during the interrupt cycle (on the same schematic, Z80 /RD is connected to RAM /OE and is pulled up). This should cause the instruction executed to be RST 38h, effectively producing the same behavior as interrupt mode 1. Genesis_MiSTer, however, doesn't seem to do anything to simulate such a condition. As far as I can tell from reading the HDL and looking at SignalTap traces, the byte for the interrupt cycle currently comes from the 68K bus interface. This means that in most cases, the ISR will never run and interrupts will remain disabled.

I can think of a few ways to approach this:

1) Change T80 to reset IM to 1 rather than 0. Higan does this. Almost certainly not accurate, because this would break 8080 compatibility on a real board.

2) Change T80 to retain IM through a non-power-on reset. MAME does this, and it makes Sonic Delta 40Mb work because a Z80 program sets IM to 1 early in the boot process.

3) Change the bus logic to put FF on the T80 data-in bus when it's not doing a read. This is sort of a hack, because it doesn't address the behavior of 68K or DMA-initiated bus cycles, even though on original consoles they presumably use the same /RD signal mentioned above. However, it does make Sonic Delta 40Mb work.

4) Change the bus logic to properly simulate Z80 open bus conditions for all initiators. I'm far from an expert on this code, but this looked pretty daunting to me. Does it make sense to attempt this without a refactor?

### PR DESCRIPTION
**Pull request by steven-a-wilson**
This is an example of a change that fixes #143 by implementing a sort of open bus behavior for non-read Z80 bus cycles. There are probably several other ways that the intended behavior could be expressed.

Marking this as a draft because I'm not familiar enough with the bus control code to be sure that it's even a particularly safe change, let alone the cleanest way of doing this, and I've only lightly tested it.

In summary, the changes are:

- Change the T80s DI connection so that it defaults to Z80_ZBUS_D instead of Z80_MBUS_D
- Set Z80_ZBUS_D to 8'hFF when in ZBUS_IDLE state and Z80_RD_N is not asserted.

Most of the second hunk is just indentation changes due to adding a begin/end; the first line with "if (Z80_RD_N)"  is the important part.

### CODE REVIEW

**Comment by steven-a-wilson** [comment]
Marked ready for review per sorgelig's comment on #143.

**Comment by jotego** [comment]
Does this fix the related issue?
